### PR TITLE
Continue trying to connect after an error or timeout

### PR DESCRIPTION
**Pull request by jjulian**
I got tired of the tickler quitting when a single request would timeout. Closing the laptop for more than a few minutes would ruin my connection to jira and confluence. And forget about overnight.

This adds a simple decay to the timer to continue trying if an error occurs - which is I think what we all want: a resilient tickler.

### CODE REVIEW

File: cms-jira-tickler.js [hunk 1]
```diff
@@ -1,33 +1,36 @@
...
+        }
+    }
 
-  var u;
-  if (window.location.host.match("confluence")) {
-    u = pageToFetch["confluence"];
-  } else {
-    u = pageToFetch["jira"];
-  }
+    var u;
```
Comment (jjulian) [inline cms-jira-tickler.js]: Fix some odd indentation.

File: cms-jira-tickler.js [hunk 2]
```diff
@@ -1,33 +1,36 @@
...
+function tickle() {
+    var xhr = new XMLHttpRequest();
+    xhr.onreadystatechange = function() {
+        if (xhr.readyState === 4) {
+            if (xhr.status === 200) {
+                console.log("tickled CMS: %o", ++count);
+                timeoutID = setTimeout(tickle, optimalDelay);
+            } else {
+                console.error("CMS tickler got non-200 response: %o", xhr.statusText);
+                delay = delay * 1.5; // simple decay
```
Comment (jjulian) [inline cms-jira-tickler.js]: 2x seems too aggressive to me, so I went with this simple decay. Anyone have a good decay formula they love?

File: cms-jira-tickler.js [hunk 3]
```diff
@@ -1,33 +1,36 @@
...
-			if (xhr.status !== 200) {
-				console.log("CMS tickler got non-200 response, giving up: %o", xhr.statusText);
-				clearInterval(intervalID);
-				return;
-			}
-			console.log("tickled CMS: %o", ++count);
-		}
-	}
+function tickle() {
+    var xhr = new XMLHttpRequest();
```
Comment (carlmjohnson) [inline cms-jira-tickler.js]: You can totally use [fetch](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API) if you want. I don't think this Chrome extension has to support IE. :-)

**Comment by jjulian** [comment]
I thought about that, but (1) the Chrome Extension docs mention xhr and
never mention fetch (https://developer.chrome.com/extensions/xhr), and (2)
there's no real benefit that I see - just potentially a cleaner conditional.

File: cms-jira-tickler.js [hunk 4]
```diff
@@ -1,33 +1,36 @@
...
+            if (xhr.status === 200) {
+                console.log("tickled CMS: %o", ++count);
+                timeoutID = setTimeout(tickle, optimalDelay);
+            } else {
+                console.error("CMS tickler got non-200 response: %o", xhr.statusText);
+                delay = delay * 1.5; // simple decay
+                timeoutID = setTimeout(tickle, delay);
+            }
+        }
+    }
```
Comment (carlmjohnson) [inline cms-jira-tickler.js]: Missing semicolon.

File: cms-jira-tickler.js [hunk 5]
```diff
@@ -1,33 +1,36 @@
...
+var optimalDelay = 120 * 1000; // 2 minutes
+var delay = optimalDelay;
 var pageToFetch = {
-  "jira": "https://jira.cms.gov/secure/ViewProfile.jspa",
-  "confluence": "https://confluence.cms.gov/users/viewmyprofile.action"
+    "jira": "https://jira.cms.gov/secure/ViewProfile.jspa",
+    "confluence": "https://confluence.cms.gov/users/viewmyprofile.action"
 };
-var intervalID = null;
+var timeoutID = null;
```
Comment (carlmjohnson) [inline cms-jira-tickler.js]: This variable is never used. I think the idea was to use it to clear the timeout, but the timeout is never cleared.

**Comment by line47** [comment]
Can this be merged?

**Comment by jjulian** [comment]
I think so. @line47 can you install it locally and give it a try?

**Comment by line47** [comment]
I downloaded and will be trying it out today 👍

**Comment by line47** [comment]
I have remained connected to both Confluence and Jira all day (8+ hours)

**Pull request merged**